plutus-tx-plugin/src/Language/PlutusTx/Compiler/Builtins.hs

```diff
@@ -260,13 +264,16 @@ defineBuiltinTerms = do
     -- Strings and chars
     do
         let term = mkDynBuiltin PLC.dynamicAppendName
-        defineBuiltinTerm 'Builtins.appendString term []
+        defineBuiltinTerm 'Builtins.appendString term [str]
```

Comment:
What does `str` mean here?

Reply:
Oh, this is just something I should have done before. These are dependencies between definitions so that PIR can generate the lets in the right order. In this case it doesn't matter much but it's just hygienic to put the dependencies in. (`appendString` needs the `string` type!).